### PR TITLE
General: Fixed quick stake amount on quick stake banner

### PR DESCRIPTION
**Pull request by standreinmcp**
Verifying now on the right amount, so the banner appears when you have over 1000 zil and dissapears when below 1000 zil.

### CODE REVIEW

File: src/components/quick-delegate-banner/quick-delegate-banner.tsx [hunk 1]
```diff
@@ -31,7 +33,7 @@ export const QuickDelegateBannerComponent = (
 
     const blockchainConfig = getBlockchain(props.blockchain);
 
-    if (!props.account) {
+    if (!props.account && !props.accountStats.chartStats) {
```
Comment (NovacAndrei) [inline src/components/quick-delegate-banner/quick-delegate-banner.tsx:36]: Just for safety, use `if (!props.account && !props.accountStats?.chartStats) {`

**Review by georgeburduhos — COMMENTED** [review]
What happens if we decide to remove Available from chart or move it to another place inside the array?

**Comment by standreinmcp** [comment]
@georgeburduhos then other functionalities will need to be changed too: for ex: I believe Available amount in Summary is implemented just the same as this one.

File: src/components/quick-delegate-banner/quick-delegate-banner.tsx [hunk 2]
```diff
@@ -48,27 +50,28 @@ export const QuickDelegateBannerComponent = (
...
 
             const minAmount = new BigNumber(clientMinAmount).plus(new BigNumber(fees));
-
             // getMinimumAmountDelegate + fees + 1%
             setMinimumAmountDelegate(minAmount.plus(minAmount.dividedBy(100)));
         })();
     }, [props.blockchain, props.account.address, props.chainId]);
 
     if (tokenConfig.ui.tokenScreenComponent === TokenScreenComponentType.DELEGATE) {
+        const avaialableChartData = props.accountStats?.chartStats.find(
```
Comment (krisboit) [inline src/components/quick-delegate-banner/quick-delegate-banner.tsx:64]: don't we have this value in redux ? I'm not happy with this solution, if we will add internationalisation `item.title` will not be `Available`. This data should be use only for display purposes.

File: src/core/blockchain/near/client.ts [hunk 3]
```diff
@@ -29,17 +30,22 @@ export class Client extends BlockchainGenericClient {
...
         try {
             const data = await new ApiClient().validators.getBalance(
                 address,
                 Blockchain.NEAR,
                 this.chainId.toString()
             );
 
-            return data?.balance?.total || new BigNumber(0);
+            return (
+                { total: data?.balance?.total, available: data?.balance?.available } || {
```
Comment (krisboit) [inline src/core/blockchain/near/client.ts:42]: `{} || ..` an empty object is not a falsy value... so `|| {..}` is useless

File: src/redux/wallets/reducer.ts [hunk 4]
```diff
@@ -26,10 +26,12 @@ import { RESET_ALL_DATA, EXTENSION_UPDATE_STATE } from '../app/actions';
 const intialState: IWalletsState = {};
 
 const newBalance = (oldBalance: any, action: any) => ({
-    value: action.data.balance ? action.data.balance : oldBalance?.value,
+    value: action.data.balance ? action.data.balance.total : oldBalance?.value,
```
Comment (krisboit) [inline src/redux/wallets/reducer.ts:29]: `value` should be `balance.available`

**Review by krisboit — CHANGES_REQUESTED** [review]

File: src/redux/wallets/reducer.ts (hunk 4 above)
Comment (standreinmcp) [reply src/redux/wallets/reducer.ts:29]: I added available 2 lines after this. so right now we have: available, total and value. Value should still be available ? 
available: action.data.balance ? action.data.balance.available : oldBalance?.available,

Comment (krisboit) [reply src/redux/wallets/reducer.ts:29]: balance.value is used in many places in our app, and in most of the cases that value represents the available amount from user's account. Here you set the total amount on `value` which will lead to misleading information.

Comment (georgeburduhos) [reply src/redux/wallets/reducer.ts:29]: If we change the `value` we must handle state migration as well
My suggestion for now: keep `value` as available and add `total` to use it where we need
@krisboit

**Review by georgeburduhos — APPROVED** [review]

File: src/core/blockchain/types/client.ts [hunk 5]
```diff
@@ -11,6 +11,11 @@ export interface IBlockInfo {
     hash?: string;
 }
 
+export interface IBalance {
+    total: BigNumber;
```
Comment (krisboit) [inline src/core/blockchain/types/client.ts:15]: There are 2 options here:
1. keep BigNumber as type here, but you have to be sure that BigNumber instances don't end up directly in redux. We would have problems on state rehydrate.
2. transform to string, redux will be fine, use BigNumber only where we need to use the amount to calculate something

File: src/core/blockchain/near/client.ts [hunk 6]
```diff
@@ -29,17 +30,22 @@ export class Client extends BlockchainGenericClient {
...
+    public async getBalance(address: string): Promise<IBalance> {
         try {
             const data = await new ApiClient().validators.getBalance(
                 address,
                 Blockchain.NEAR,
                 this.chainId.toString()
             );
-
-            return data?.balance?.total || new BigNumber(0);
+            if (data.balance.total !== undefined && data.balance.available !== undefined)
```
Comment (krisboit) [inline src/core/blockchain/near/client.ts:40]: you don't really need an if here, you can do something like:
```ts
return {
  total: data?.balance?.total || 0
  available: ...
}
```

Comment (krisboit) [reply src/core/blockchain/near/client.ts:40]: the types may not be ok in my example

File: src/core/blockchain/types/client.ts [hunk 7]
```diff
@@ -31,7 +36,7 @@ export abstract class BlockchainGenericClient {
         this.http = new HttpClient(url);
     }
 
-    public abstract getBalance(address: string): Promise<BigNumber>;
+    public abstract getBalance(address: string): Promise<IBalance | BigNumber>;
```
Comment (krisboit) [inline src/core/blockchain/types/client.ts:39]: I'm not ok with the `IBalance | BigNumber`, it should be one or the other.

File: src/components/quick-delegate-banner/quick-delegate-banner.tsx [hunk 8]
```diff
@@ -31,7 +33,7 @@ export const QuickDelegateBannerComponent = (
 
     const blockchainConfig = getBlockchain(props.blockchain);
 
-    if (!props.account) {
+    if (!props.account && !props.accountStats?.chartStats) {
```
Comment (krisboit) [inline src/components/quick-delegate-banner/quick-delegate-banner.tsx:36]: is there any dependency on chartStats for this banner ?

**Review by krisboit — CHANGES_REQUESTED** [review]

File: src/components/quick-delegate-banner/quick-delegate-banner.tsx (hunk 8 above)
Comment (standreinmcp) [reply src/components/quick-delegate-banner/quick-delegate-banner.tsx:36]: not anymore

File: src/core/blockchain/types/client.ts (hunk 5 above)
Comment (standreinmcp) [reply src/core/blockchain/types/client.ts:15]: Before, value was as a BigNumber, I don't think it gets to redux as a BigNumber but I'll double check

File: src/components/quick-delegate-banner/quick-delegate-banner.tsx (hunk 8 above)
Comment (krisboit) [reply src/components/quick-delegate-banner/quick-delegate-banner.tsx:36]: ok, remove it :)

File: src/core/blockchain/types/client.ts (hunk 5 above)
Comment (krisboit) [reply src/core/blockchain/types/client.ts:15]: please do

Comment (standreinmcp) [reply src/core/blockchain/types/client.ts:15]: It's fine, they are all converted to string before

File: src/core/blockchain/ethereum/client.ts [hunk 9]
```diff
@@ -24,9 +25,9 @@ export class Client extends BlockchainGenericClient {
         this.utils = new ClientUtils(this);
     }
 
-    public getBalance(address: string): Promise<BigNumber> {
+    public getBalance(address: string): Promise<IBalance> {
         return this.http.jsonRpc('eth_getBalance', [fixEthAddress(address), 'latest']).then(res => {
-            return new BigNumber(res.result, 16);
+            return { total: new BigNumber(res.result, 16), available: new BigNumber(0) };
```
Comment (krisboit) [inline src/core/blockchain/ethereum/client.ts:30]: available shoudn't be 0. On ETH total = available

**Review by krisboit — CHANGES_REQUESTED** [review]

**Review by krisboit — APPROVED** [review]

**Pull request merged**